module-4/backend/week_three.md

```diff
@@ -1,19 +1,33 @@
 ## Week Three - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. At a high level, what is Node?
+  - Node is essentially a way to run JavaScript on the server side using Google's V8 JavaScript engine.
 2. What is Express? What is Express similar to in the Ruby world?
+  - Express is a web application framework for JavaScript. I would probably compare it to Sinatra in the way that it is structured.
 3. How do we setup a route when creating an API with Node and Express? Please provide a code snippet.
+  - ```
+    app.get('/', function(request, response) {
+        response.send('Welcome home.')
+      })
+    ```
 4. What do we use Knex for?
+  - Knex is used for writing migrations and building SQL queries in express.
 5. How could you organize your code to follow the MVC design pattern in your Quantified Self project?
+  - By kind of modeling the server.js file as a router, and then abstracting code dealing with retrieving information from the database into models, code dealing with requesting and passing that information to the controller level, and I'm not quite sure how the view fits here with Quantified Self since it was in a separate app and I've always envisioned the view layer as more a template that produces html rather than an actual html page with JS.
```

Comment:
you're right! in this case...we don't have views. we're simply returning data to be consumed by a third party 👍